client/src/hooks/useGetProjectDetails.js

```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+import { useAuthContext } from "./useAuthContext";
+
+export const useGetProjectDetails = () =>{
+    const { user } = useAuthContext();
+    const [error, setError] = useState(null);
+    const [isLoading, setIsLoading] = useState(null);
+
+    const getProjectDetails = async (listing_id) =>{
+        setIsLoading(true);
+        setError(null);
+        const response = await fetch('api/user/listings/'+listing_id, {
```

Comment:
Please consider using a template literal instead of string concatenation:  `/api/user/listings/${listing_id}`